S2VX.Game/Editor/NotesTimeline.cs

```diff
@@ -214,20 +235,30 @@ public class NotesTimeline : CompositeDrawable {
             }
         }
 
-        protected override void Update() {
-            TickBar.Clear();
-            TickBar.Add(new RelativeBox {
-                Colour = Color4.White,
-                Height = TimelineHeight / 3.5f,
-            });
+        private void AddVisibleNotes() {
+            var lowerBound = Story.GameTime - SectionLength * 1000 / 2;
+            var upperBound = Story.GameTime + SectionLength * 1000 / 2;
+            foreach (var note in Story.Notes.Children) {
+                if (lowerBound <= note.EndTime && note.EndTime <= upperBound) {
+                    var relativePosition = (note.EndTime - lowerBound) / (SectionLength * 1000);
+                    var visibleNote = new RelativeBox {
+                        Name = "NoteBox",
+                        Colour = Color4.White.Opacity(0.727f),
+                        Width = TimelineNoteWidth,
+                        Height = TimelineNoteHeight,
+                        X = (float)relativePosition,
+                        Y = 0.2f,
+                        Anchor = Anchor.TopLeft,
+                    };
+                    NoteToTimelineNote[note] = visibleNote;
+                    TickBar.Add(visibleNote);
+                }
+            }
+        }
 
-            TickBar.Add(new RelativeBox {
-                Colour = Color4.White,
-                Width = TimelineWidth / 350,
-                Height = 0.8f,
-                Anchor = Anchor.TopCentre,
-                Y = 0.1f,
-            });
+        protected override void Update() {
+            NoteToTimelineNote.Clear();
+            TickBar.RemoveAll(item => item.Name == "Tick" || item.Name == "NoteBox");
```

Comment:
Doing this kind of Name check magic seems kind of smelly to me. How hard would it be to have separate containers containing these and then just clearing those?

Reply:
I could probably avoid doing the name checks if I made two separate Tickbars with the same dimensions to hold the Ticks & NoteBoxes separately from the TickBar & MiddleTick. I also do the name checks in SelectToolState.cs to remove note selections on the timeline, so I would need 3 separate tickbars to hold these different items.

Reply:
What's wrong with putting a placeholder container into Tickbar just for the Ticks and NoteBoxes?

Reply:
Nvm I didn't think of that lol. I'll try doing that Rn.